### PR TITLE
Add weighted multi-symbol portfolio simulation

### PR DESCRIPTION
**Pull request by michaelchu**
## Summary
- Add `simulate_portfolio()` function for weighted multi-symbol backtesting
- Each leg runs independently via `simulate()` with proportional capital allocation
- Combined trade log, equity curve, and portfolio-level risk metrics (Sharpe, Sortino, max drawdown, VaR, etc.)
- Per-leg breakdowns available via `PortfolioResult.leg_results`
- 32 new tests covering validation, trade log merging, equity curves, summary metrics, and edge cases

## Test plan
- [x] `uv run pytest tests/test_simulator_portfolio.py -v` — all 32 tests pass
- [x] `uv run pytest tests/ -v` — full suite (1077 tests) passes
- [x] `uv run ruff format --check optopsy/ tests/` — clean
- [x] `uv run ruff check optopsy/ tests/` — clean
- [x] Pre-commit hooks (ruff format, ruff check, ty) pass

### CODE REVIEW

File: optopsy/simulator.py [hunk 1]
```diff
@@ -733,3 +750,150 @@ def simulate(
...
+        name = leg.get("name") or getattr(strategy, "__name__", f"leg{i}")
+
+        # Deduplicate names
+        if name in seen_names:
+            name = f"{name}_{i}"
+        seen_names.add(name)
+
+        leg_capital = capital * weight
+        leg_capitals[name] = leg_capital
+
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/simulator.py:837]: Fixed in 48f4cdb — `leg_capitals` is now passed to `_combine_equity_curves()` where it seeds each leg's starting capital into the portfolio equity curve.

File: optopsy/simulator.py [hunk 2]
```diff
@@ -733,3 +750,150 @@ def simulate(
...
+        curve = result.equity_curve.copy()
+        curve.index = pd.to_datetime(curve.index)
+        # Resample to daily; use last value per day, forward-fill gaps
+        daily = curve.resample("D").last().ffill()
+        daily_curves[name] = daily
+
+    if not daily_curves:
+        return pd.Series(dtype=float, name="equity")
+
+    combined = pd.DataFrame(daily_curves).ffill().sum(axis=1)
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/simulator.py]: Fixed in 48f4cdb — `_combine_equity_curves()` now accepts `leg_capitals` and seeds each leg with its starting capital from portfolio start date. Legs that never trade show as a flat line at their allocated capital. Added test `test_idle_cash_in_equity_curve` to verify.

File: optopsy/simulator.py [hunk 3]
```diff
@@ -733,3 +750,150 @@ def simulate(
...
+
+        leg_capital = capital * weight
+        leg_capitals[name] = leg_capital
+
+        # Build kwargs for simulate(): everything except reserved keys
+        sim_kwargs: dict[str, Any] = {
+            k: v for k, v in leg.items() if k not in _PORTFOLIO_RESERVED_KEYS
+        }
+        sim_kwargs["capital"] = leg_capital
+
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/simulator.py:843]: Fixed in 48f4cdb — weights are now normalized by `total_weight` after validation so allocated capital always sums exactly to `capital`. Added test `test_weights_normalized` to verify.

File: optopsy/simulator.py [hunk 4]
```diff
@@ -733,3 +750,187 @@ def simulate(
...
+    if not all_logs:
+        cols = _TRADE_LOG_COLUMNS + ["leg"]
+        return pd.DataFrame(columns=cols)
+
+    combined = pd.concat(all_logs, ignore_index=True)
+    combined = combined.sort_values("entry_date").reset_index(drop=True)
+    combined["trade_id"] = np.arange(1, len(combined) + 1)
+    combined["cumulative_pnl"] = combined["realized_pnl"].cumsum()
+    combined["equity"] = capital + combined["cumulative_pnl"]
+    return combined
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/simulator.py:888]: Fixed in 40983ef — `_combine_trade_logs()` now sorts by `exit_date` (with `entry_date` and `leg` as deterministic tie-breakers) so cumulative P&L tracks realization order. Updated the corresponding test.

File: optopsy/simulator.py [hunk 5]
```diff
@@ -733,3 +750,187 @@ def simulate(
...
+
+    for i, leg in enumerate(legs):
+        data = leg["data"]
+        strategy = leg["strategy"]
+        weight = leg["weight"] * norm_factor
+        name = leg.get("name") or getattr(strategy, "__name__", f"leg{i}")
+
+        # Deduplicate names
+        if name in seen_names:
+            name = f"{name}_{i}"
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply optopsy/simulator.py]: Fixed in 40983ef — name deduplication now loops with an incrementing suffix until it finds a unique name, avoiding collisions with explicitly named legs. Added test `test_dedup_avoids_collision_with_explicit_name`.

File: tests/test_simulator_portfolio.py [hunk 6]
```diff
@@ -0,0 +1,811 @@
...
+        )
+
+        # Per-leg capital should sum to total capital (not 100,500)
+        a_cap = (
+            result.leg_results["a"].summary["total_pnl"]
+            / result.leg_results["a"].summary["total_return"]
+        )
+        b_cap = (
+            result.leg_results["b"].summary["total_pnl"]
+            / result.leg_results["b"].summary["total_return"]
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_simulator_portfolio.py]: Fixed in 8222c36 — added zero-division guard. The test now asserts total_return is non-zero before dividing, and falls back to asserting 0.0 for zero P&L cases.

File: tests/test_simulator_portfolio.py [hunk 7]
```diff
@@ -0,0 +1,811 @@
...
+        if spy_result.summary["total_trades"] > 0:
+            expected_return = spy_result.summary["total_pnl"] / 70_000
+            assert spy_result.summary["total_return"] == pytest.approx(
+                expected_return, rel=1e-6
+            )
+        if qqq_result.summary["total_trades"] > 0:
+            expected_return = qqq_result.summary["total_pnl"] / 30_000
+            assert qqq_result.summary["total_return"] == pytest.approx(
+                expected_return, rel=1e-6
+            )
```
[the remark this answers is not in the document]
Comment (michaelchu) [reply tests/test_simulator_portfolio.py]: Fixed in 8222c36 — added explicit zero-P&L guards in test_capital_allocation. Non-zero P&L paths divide as before; zero P&L asserts total_return is also 0.0.
